### PR TITLE
remove outdated todos

### PR DESCRIPTION
**Pull request by idegtiarenko**

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceComputer.java [hunk 1]
```diff
@@ -188,14 +188,6 @@ public DesiredBalance compute(
             }
         }
 
-        // TODO must also bypass ResizeAllocationDecider
-        // TODO must also bypass RestoreInProgressAllocationDecider
-        // TODO what about delayed allocation?
```
Comment (idegtiarenko) [inline server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceComputer.java:193]: Delayed allocations are handled elsewhere in ReplicaShardAllocator

File: server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceComputer.java [hunk 2]
```diff
@@ -188,14 +188,6 @@ public DesiredBalance compute(
...
-
-        // TODO consider also whether to unassign any shards that cannot remain on their current nodes so that the desired balance reflects
-        // the actual desired state of the cluster. But this would mean that a REPLACE shutdown needs special handling at reconciliation
-        // time too. But maybe it needs special handling anyway since reconciliation also tries to respect allocation rules.
```
Comment (idegtiarenko) [inline server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceComputer.java:197]: This should be handled in both compute and reconcile balance due to `org.elasticsearch.cluster.routing.allocation.decider.AllocationDecider#canRemain` calls

File: server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceComputer.java [hunk 3]
```diff
@@ -188,14 +188,6 @@ public DesiredBalance compute(
             }
         }
 
-        // TODO must also bypass ResizeAllocationDecider
```
Comment (idegtiarenko) [inline server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceComputer.java:191]: Should be fixed by https://github.com/elastic/elasticsearch/pull/87954

File: server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceComputer.java [hunk 4]
```diff
@@ -188,14 +188,6 @@ public DesiredBalance compute(
...
-        // TODO must also bypass RestoreInProgressAllocationDecider
```
Comment (idegtiarenko) [inline server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceComputer.java:192]: Not sure why this is mentioned here. No relevant test failures recorded

**Review by DaveCTurner — APPROVED** [review]
LGTM

**Pull request merged**